### ISSUE TITLE
Increase flexibility of the installer

### ISSUE THREAD

**Issue by Rafiot** [certtools/intelmq#181]
This is not really a bug, just me looking at the installer and taking some notes.. :) 

- [x] I like being able to install projects in my home directory, I'm not sure how the project will like it, so I stick to the /opt/ way for now, and will look at it later (seems that the only hard coded paths are the config files, which makes total sense). Same as running an install script as root, me not liking it. (related to 2.)
- [x] I always install python projects in a virtualenv (so I do not pollute the rest of the system with pip packages). And I can have multiple projects running at once on the same bare system without having random issues with updates. That seems to work, updates to come in case it doesn't.
- [x] Upgrade: a requirements.txt file with the list of dependencies would a be nice to have, so it is possible to update them (and only them) more easily.
- [x] Uninstall: that should never be done on a bare machine with other services installed, or I'm pretty sure they will be broken.
- [x] I assumes you have one single redis instance installed system wide, on the default port. In our infrastructure, we will have tons of redis instances running on the same host. Would be a good idea to not start polluting redis instances that may be running on the same host by mistake.

**Comment by SYNchroACK** [comment]
Hi @Rafiot , if you have time, check the new version that is under developments.

https://github.com/certtools/intelmq-beta

**Comment by Rafiot** [comment]
Hi @SYNchroACK, thank you, I will. 
Just a practical question: why not just using a dev branch on the main project for the future release? it would be much easier to follow the differences.

**Comment by aaronkaplan** [comment]
On May 26, 2015, at 8:23 AM, Raphaël Vinot <notifications@github.com> wrote:

> Hi @SYNchroACK, thank you, I will. 
> Just a practical question: why not just using a dev branch on the main project for the future release? it would be much easier to follow the differences.
> 

Yeah, I asked the same. Seems the -beta repo will be deleted after successful merging into the main master tree.

**Comment by Rafiot** [comment]
Well, that's why branches are there :)

**Comment by aaronkaplan** [comment]
On May 26, 2015, at 11:26 AM, Raphaël Vinot <notifications@github.com> wrote:

> Well, that's why branches are there :)
> 

I agree..

**Comment by SYNchroACK** [comment]
@Rafiot , beta version has zmq support .... lib/pipeline.py 
intelmqctl etc need some improvements to support it...

**Comment by Rafiot** [comment]
Ok. I'm not sure it is interesting for you, but his is the way I implemented the redis & ZMQ support for AIL: https://github.com/CIRCL/AIL-framework/blob/master/bin/Helper.py

**Comment by SYNchroACK** [comment]
Thanks

**Comment by SYNchroACK** [comment]
```
This is not really a bug, just me looking at the installer and taking some notes.. :) 

1. I like being able to install projects in my home directory, I'm not sure how the project will like it, so I stick to the /opt/ way for now, and will look at it later (seems that the only hard coded paths are the config files, which makes total sense). Same as running an install script as root, me not liking it. (related to 2.)
```

In the new version, v1.0-beta branch, the __init__.py file in intelmq root is the only place where the PATHs are defined. In your custom installation you need to change it there and also the setup.py. If you see an opportunity to improve it, please, send a pull req. :)

```
2. I always install python projects in a virtualenv (so I do not pollute the rest of the system with pip packages). And I can have multiple projects running at once on the same bare system without having random issues with updates. That seems to work, updates to come in case it doesn't.
```
Thanks for the feedback.

```
3. Upgrade: a requirements.txt file with the list of dependencies would a be nice to have, so it is possible to update them (and only them) more easily.
```
v1.0-beta branch it has REQUIREMENTS file since you suggest the idea ;)

```
4. Uninstall: that should never be done on a bare machine with other services installed, or I'm pretty sure they will be broken.
```
:( dont know how to solve it in case of user need to uninstall without broke anything.

```
5. It assumes you have one single redis instance installed system wide, on the default port. In our infrastructure, we will have tons of redis instances running on the same host. Would be a good idea to not start polluting redis instances that may be running on the same host by mistake.
```
v1.0-beta branch we are trying to improve it.

**Comment by SYNchroACK** [comment]
> > This is not really a bug, just me looking at the installer and taking some notes.. :) 

.

> > 1 - I like being able to install projects in my home directory, I'm not sure how the project will like it, so I stick to the /opt/ way for now, and will look at it later (seems that the only hard coded paths are the config files,
> > which makes total sense). Same as running an install script as root, me not liking it. (related to 2.)
> 
> In the new version, v1.0-beta branch, the __init__.py file in intelmq root is the only place where the PATHs are defined. In your custom installation you need to change it there and also the setup.py. If you see an opportunity to 
> improve it, please, send a pull req. :)

v1.0-beta branch was merged with master branch. :)


> > 2 - I always install python projects in a virtualenv (so I do not pollute the rest of the system with pip packages). And I can have multiple projects running at once on the same bare system without having random issues with updates. 
> > That seems to work, updates to come in case it doesn't.
> 
> Thanks for the feedback.

.

> > 3 - Upgrade: a requirements.txt file with the list of dependencies would a be nice to have, so it is possible to update them (and only them) more easily.
> 
> v1.0-beta branch it has REQUIREMENTS file since you suggest the idea ;)

v1.0-beta branch was merged with master branch. :)


> > 4 - Uninstall: that should never be done on a bare machine with other services installed, or I'm pretty sure they will be broken.
> 
> :( dont know how to solve it in case of user need to uninstall without broke anything.

Until now, any news about it. :(


> > 5 - It assumes you have one single redis instance installed system wide, on the default port. In our infrastructure, we will have tons of redis instances running on the same host. Would be a good idea to not start polluting redis 
> > instances that may be running on the same host by mistake.
> 
> v1.0-beta branch we are trying to improve it.

Master branch has now the correct approach. (I hope :P)

**Comment by sebix** [comment]
Most issues have been solved in #448 and #391 and #451 

Still unresolved: usage of multiple intelmq-instances on one redis-instance

**Comment by sebix** [comment]
I think i misunderstood the last point. It's about configuring the redis port (which has been implemented "long" ago).

Closing this, everything is done. If not, please reopen or open another issue and explain in more detail.
